src/ResourceManager/Sql/Commands.Sql/SqlDatabaseAgent/Services/TargetGroup/AzureSqlDatabaseAgentTargetGroupCommunicator.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+        /// <summary>
+        /// Gets the associated Target group associated to the Azure SQL Database Agent
+        /// </summary>
+        /// <param name="resourceGroupName"></param>
+        /// <param name="serverName"></param>
+        /// <param name="agentName"></param>
+        /// <param name="targetGroupName"></param>
+        /// <returns>The target group belonging to specified agent</returns>
+        public Management.Sql.Models.JobTargetGroup Get(
+            string resourceGroupName,
+            string serverName,
+            string agentName,
+            string targetGroupName)
+        {
+            return GetCurrentSqlClient().JobTargetGroups.Get(resourceGroupName, serverName, agentName, targetGroupName);
+        }
+
+        /// <summary>
+        /// Lists the target groups associated to the Azure SQL Database Agent.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server name</param>
+        /// <param name="agentName">The agent name</param>
+        /// <returns>A list of target groups belonging to specified agent</returns>
+        public IPage<Management.Sql.Models.JobTargetGroup> List(
+            string resourceGroupName,
+            string serverName,
+            string agentName)
+        {
+            return GetCurrentSqlClient().JobTargetGroups.ListByAgent(resourceGroupName, serverName, agentName);
+        }
+
+        /// <summary>
+        /// Deletes the target groups associated to the agent.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server name</param>
+        /// <param name="agentName">The agent name</param>
+        /// <param name="targetGroupName">The target group name</param>
+        public void Remove(
+            string resourceGroupName,
+            string serverName,
+            string agentName,
+            string targetGroupName)
+        {
+            GetCurrentSqlClient().JobTargetGroups.Delete(resourceGroupName, serverName, agentName, targetGroupName);
+        }
+
+        /// <summary>
+        /// Retrieve the SQL Management client for the currently selected subscription, adding the session and request
+        /// id tracing headers for the current cmdlet invocation.
+        /// </summary>
+        /// <returns>The SQL Management client for the currently selected subscription.</returns>
+        private SqlManagementClient GetCurrentSqlClient()
+        {
+            // Get the SQL management client for the current subscription
+            // Note: client is not cached in static field because that causes ObjectDisposedException in functional tests.
+            var sqlClient = AzureSession.Instance.ClientFactory.CreateArmClient<SqlManagementClient>(Context, AzureEnvironment.Endpoint.ResourceManager);
+            return sqlClient;
+        }
+    }
+}
```

Comment:
You will need to generate help files: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md

You will also need to update the packages.config for both commands.sql and commands.sql.test: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/packages.config